substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -673,6 +674,12 @@ public int headerSize() {
             return getObjectLayout().getFirstFieldOffset();
         }
 
+        @Override
+        public String loaderName() {
+
+            return UniqueShortNameProvider.singleton().uniqueShortName(hostedType.getJavaClass().getClassLoader());
```

Comment:
This gave me a heart attack for a moment. I though you are trying to use the FQ class-name of the classloader. But you have overloaded the `uniqueShortName` method name to have a variant that takes a loader :sweat_smile: 

It might be good not to use `uniqueShortName` for that but a more explicit `uniqueShortLoaderName`.

Reply:
Yes, indeed. Much healthier.